alibi/explainers/pd_variance.py

```diff
@@ -0,0 +1,660 @@
+            raise ValueError(f"Unknown mode. Received ``method={method}``. "
+                             f"Accepted `method` names are: {get_options_string(Method)}")
+
+        # get number of features
+        n_features = X.shape[1]
+
+        # construct features if not provided based on the method
+        if features is None:
+            features = list(range(n_features))
+
+            if method == Method.INTERACTION:
+                features = list(combinations(features, 2))
+
+        # compute partial dependence functions
+        params = {
+            'X': X,
+            'features': features,
+            'percentiles': percentiles,
+            'grid_resolution': grid_resolution,
+            'grid_points': grid_points
+        }
+
+        if not isinstance(self.pd_explainer.predictor, BaseEstimator):
+            params.update({'kind': Kind.AVERAGE.value})  # type: ignore[dict-item]
+
+        # compute partial dependence for each feature
+        pd_explanation = self.pd_explainer.explain(**params)  # type: ignore[arg-type]
+
+        if method == Method.IMPORTANCE:
+            if not all([isinstance(f, numbers.Integral) for f in features]):
+                raise ValueError(f"For ``method='{Method.IMPORTANCE.value}'`` all features must be integers.")
+            # compute feature importance
+            buffers = self._compute_feature_importance(pd_explanation=pd_explanation, features=features)
+        else:
+            if not all([isinstance(fs, tuple) and len(fs) for fs in features]):
+                raise ValueError(f"For ``method='{Method.INTERACTION.value}'`` all features must be "
+                                 f"tuples of length 2.")
+            # compute feature interaction
+            buffers = self._compute_feature_interaction(pd_explanation=pd_explanation,
+                                                        features=features)  # type: ignore[arg-type]
+
+        # update meta and return built explanation
+        self.meta['params'].update(self.pd_explainer.meta['params'])
+        self.meta['params'].update({'method': method})
+        return self._build_explanation(buffers=buffers)
+
+    def _compute_pd_variance(self, features: List[int], pd_values: List[np.ndarray]) -> np.ndarray:
+        """
+        Computes the PD variance along the final axis for all the features.
+
+        Parameters
+        ----------
+        features
+            See :py:meth:`alibi.explainers.pd_variance.PartialDependenceVariance.explain`.
+        pd_values
+            List of length `F` containing the PD values for each feature in `features`. Each PD value is an array
+            with the shape `T x N1 ... x Nk` where `T` is the number of targets and `Ni` is the number of feature
+            values along the `i` axis.
+
+        Returns
+        -------
+        An array of size `F x T x N1 x ... N(k-1)`, where `F` is the number of explained features, `T` is the number
```

Comment:
Not sure I understand why it goes up to `N(k-1)` not `N(k)`.

Reply:
Because we compute the variance along the `Nk` and thus will be removed. For a 2D matrix, if we compute the variance along the second axis, then the result will be 1D.